.github/workflows/workflow.yaml

```diff
@@ -50,13 +53,35 @@ jobs:
           	EOF
           fi
 
+      - name: Run tests (bzlmod)
+        if: matrix.bazel_mode == 'module'
+        working-directory: ./tests
+        run: |
           echo "USE_BAZEL_VERSION=${{ matrix.version }}" > .bazeliskrc
 
-      - name: Build & Test
-        run: |
+          # Make the configs also available in the tests directory.
+          cp ../.bazelrc.local .
+          cp ../.bazelrc .
+
           if [[ ${{ runner.os }} == Windows ]]; then
             # On Windows `//...` expands to `/...`.
-            BAZEL_SH='C:\msys64\usr\bin\bash.exe' bazel test ///...
+            BAZEL_SH='C:\msys64\usr\bin\bash.exe' bazel test  ///...
+          else
+            bazel test  //...
+          fi
+
+      - name: Run tests (workspace)
+        if: matrix.bazel_mode == 'workspace'
+        working-directory: ./tests
+        run: |
+          echo "USE_BAZEL_VERSION=${{ matrix.version }}" > .bazeliskrc
+
+          # Make the configs also available in the tests directory.
+          cp ../.bazelrc.local .
+          cp ../.bazelrc .
+
+          if [[ ${{ runner.os }} == Windows ]]; then
+            BAZEL_SH='C:\msys64\usr\bin\bash.exe' bazel test @rules_sh//... @rules_sh_tests//...
           else
-            bazel test //...
+            bazel test @rules_sh//... @rules_sh_tests//...
```

Comment:
This is more naturally done in two separate steps. One of them runs in the repository root and invokes `bazel test //...` to test the `rules_sh`, and the other runs in `tests/` and also just invokes `bazel test //...`.

Reply:
If we go with two commands won't it cancel the `working-directory: ./tests` parameter that matches the invocation for the bzlmod case?

Reply:
I don't think I understand. How would it cancel?

I could imagine a couple ways to achieve this, but I may well be missing something
1. ```
    - name: Run tests (workspace)
      if: matrix.bazel_mode == 'workspace'
      run: |
        ...
        bazel test //...
        (cd tests && bazel test //...)
    ```
3.
    ```
    - name: Main module (workspace)
      if: matrix.bazel_mode == 'workspace'
      run: |
        ...
        bazel test //...
    - name: Test module (workspace)
      if: matrix.bazel_mode == 'workspace'
      working-directory: ./tests
      run: |
        ...
        bazel test //...
    ```
4.
    ```
    matrix:
      ...
      bazel_mode: [workspace, module]
      workdir: [., tests]
    
    ...
    
    - name: Run tests
      working-directory: ${{ matrix.workdir }}
      run: |
        ...
        bazel test //...
    ```

Reply:
I meant that it would be extra commands/configs and it will diverge from the bzlmod job config. e.g 1. calls `cd tests` where in bzlmod is using the `working-directory`, 2 & 3 add a new job.

Reply:
Ideally, both would be doing both. I.e. ideally we run `bazel test //...` in both `./` and `tests/` under both `WORKSPACE` and `bzlmod`. However, we can't run the Stardoc tests under `./docs` with bzlmod enabled at the moment, because Stardoc currently doesn't work in bzlmod mode. In the future, whenever that's resolved, we would also want to test `./` under bzlmod. So, this should be seen as temporary difference between the modes. The configuration could reflect that.

> 2 & 3 add a new job.

3 adds a new job, 2 only adds a new step.

Reply:
I've updated the workspace tests to use the `//...` convention by `cd`ing into the tests directory.